IntelliTect.TestTools.Selenate/GoogleSearch/GoogleBrowser.cs

```diff
@@ -8,17 +10,32 @@ public class GoogleBrowser : Browser
         public GoogleBrowser(BrowserType browser) : base(browser)
         {
         }
+        
+        public WebElement FindElement(By by)
+        {
+            return new WebElement(Find(by).ConfigureAwait(false).GetAwaiter().GetResult(), Driver);
+        }
+
+        // Instead of doing this, I should just do a separate project demonstrating async usage on something more complex than a Google search
+        public async Task<WebElement> FindElementAsync(By by)
+        {
+            return new WebElement(await Find(by), Driver);
+        }
 
         // When running tests in succession, Google sometimes refuses the connection.
         // Refresh to kick it into gear.
-        public WebElement FindElement(By by)
+        private async Task<IWebElement> Find(By by)
         {
-            WebElement result = base.FindElement(by);
-            if(!result.Initialized)
+            try
+            {
+                return await base.FindElement(by);
+            }
+            catch (AggregateException ae)
             {
+                var test = ae;
```

Comment:
Ah, yep. I was having an issue getting the catch to work. Turns out it was because I had other issues and something other than an AggregateException was being thrown. :) Removed in next commit.